src/observable/ConnectableObservable.ts

```diff
@@ -17,28 +17,35 @@ export class ConnectableObservable<T> extends Observable<T> {
     return this._getSubject().subscribe(subscriber);
   }
 
-  _getSubject() {
-    const subject = this.subject;
+  protected _getSubject() {
+    const subject = this._subject;
     if (subject && !subject.isUnsubscribed) {
       return subject;
     }
-    return (this.subject = this.subjectFactory());
+    return (this._subject = this.subjectFactory());
   }
 
   connect(): Subscription {
     const source = this.source;
-    let subscription = this.subscription;
+    let subscription = this._subscription;
     if (subscription && !subscription.isUnsubscribed) {
       return subscription;
     }
     subscription = source.subscribe(this._getSubject());
     subscription.add(new ConnectableSubscription(this));
-    return (this.subscription = subscription);
+    return (this._subscription = subscription);
   }
 
   refCount(): Observable<T> {
     return new RefCountObservable(this);
   }
+
+  // This method is opened for `ConnectableSubscription`.
+  // Not to call from others.
+  _closeSubscription(): void {
```

Comment:
@kwonoj 

I think we should keep `_` for this method. `internal` modifier is not yet. I agree to use `private` or `protected` instead of `_`.

But all public field of all exported object is published as type definitions which is used from some autocomplete system. We need to some marker style until `internal` modifier or slimier one to mark it is not a public callable member.